numba/tests/test_np_functions.py

```diff
@@ -2503,6 +2507,43 @@ def check_pass_through(jitted, expect_same, params):
                 else:
                     check_pass_through(cfunc, True, params)
 
+    def test_repeat(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a, repeats):
+            np.testing.assert_equal(pyfunc(a, repeats), cfunc(a, repeats))
+
+        target_inputs = [
+            np.ones(1),
+            np.arange(1000),
+            np.array([[0, 1], [2, 3]]),
+            np.array([]),
+            np.array([[], []]),
+            1,
+            1.0,
+            True,
+            1j,
+            [0, 1, 2],
+            (0, 1, 2),
+        ]
+        for i in target_inputs:
+            check(i, repeats=0)
+            check(i, repeats=1)
+            check(i, repeats=2)
+
+    def test_repeat_exception(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+        self.disable_leak_check()
+
+        with self.assertRaises(ValueError):
```

Comment:
From where is the value error being raised? I think it might be coming from the allocator, which leads to there being no hint to the user in the backtrace about where the problem arose. It may be better to put the check in the `np.repeats` implementation.